src/commands/timeout.ts

```diff
@@ -0,0 +1,160 @@
+//Template: https://github.com/Costpap/CostBot/blob/main/src/commands/kick.ts
+
+//imports
+import { ChatInputCommandInteraction, PermissionFlagsBits, SlashCommandBuilder } from 'discord.js';
+import { notifyUser } from '../utils/misc';
+import ms from 'ms';
+
+export default {
+    data: new SlashCommandBuilder()
+        .setName('timeout')
+        .setDescription('Manages the timeout of a selected user.')
+        .addSubcommand((subcommand) =>
+            subcommand
+                .setName('add')
+                .setDescription('Times out the selected user.')
+                .addUserOption((option) => option.setName('user').setDescription('User to time out').setRequired(true))
+                .addStringOption((option) =>
+                    option
+                        .setName('duration')
+                        .setDescription('Duration of the timeout (e.g. 10sec/60m/2.5 hrs)')
+                        .setRequired(true),
+                )
+                .addStringOption((option) =>
+                    option.setName('reason').setDescription('Reason for timing out the user').setRequired(false),
+                )
+                .addBooleanOption((option) =>
+                    option.setName('silent').setDescription('Whether or not to DM the user').setRequired(false),
+                ),
+        )
+        .addSubcommand((subcommand) =>
+            subcommand
+                .setName('remove')
+                .setDescription('Removes timeout from a user.')
+                .addUserOption((option) =>
+                    option.setName('user').setDescription('User to remove the timeout from').setRequired(true),
+                )
+                .addStringOption((option) =>
+                    option.setName('reason').setDescription('Reason for removing the timeout').setRequired(false),
+                ),
+        ),
+    run: async (interaction: ChatInputCommandInteraction) => {
+        // Typeguard in order to ensure having access to ChatInputCommand interaction options.
+        if (!interaction.isChatInputCommand()) return;
+
+        if (interaction.inGuild() === false) {
+            return interaction.reply({ content: "❌ I can't execute this command inside DMs!", ephemeral: true });
+        }
+
+        if (!interaction.guild.members.me.permissions.has(PermissionFlagsBits.ModerateMembers)) {
+            return interaction.reply({
+                content: '❌ Sorry, I need the `Moderate Members` in order to execute this command.',
```

Comment:
Please change `Moderate Members` to `Timeout Members`, in accordance with what is shown in Discord's UI.
```suggestion
                content: '❌ Sorry, I need the `Moderate Members` in order to execute this command.',
```

Reply:
I can make this change, however, I think it is shown as "Moderate Members" even on the UI. Photo attached. 
![image](https://user-images.githubusercontent.com/90683098/193907925-b60e8388-ba22-4650-90ef-c177225fe557.png)

Reply:
Where did you get that screenshot from? Normally, when using the en-US locale, the permission tied to timing out members is "Timeout Members". cc my image below.

![image](https://user-images.githubusercontent.com/37239383/194103730-82262552-ab93-4247-b374-9b2a2149fcea.png)

Reply:
This is from the English, UK client. However, I think it will be later changed to Timeout Members there as well, as it would makes sense. Maybe if we are lucky, thel will change the PermissionBitField to TimeoutMembers as well in the future.

I'll go with Timeout Members in the text.